component/core/src/main/java/org/exoplatform/social/core/jpa/search/ProfileSearchConnector.java

```diff
@@ -534,6 +543,45 @@ private String buildExpression(ProfileFilter filter) {
     return esExp.toString();
   }
 
+  private String buildAdvancedFilterExpression(ProfileFilter filter) {
+    StringBuilder esExp = new StringBuilder();
+    esExp.append("( ");
+    Map<String,String> settings = new HashMap<>();
+    settings = filter.getProfileSettings();
+
+    int index = 0 ;
+    String[]  splitedValue = new String[0];
```

Comment:
not needed IMO since you can direcly instanciate it into for loop

Reply:
fixed